core/domain/user_domain.py

```diff
@@ -1036,15 +1096,28 @@ def validate(self):
                 'have duplicate values, found: %s' % (
                     self.can_review_voiceover_for_language_codes))
 
-        if not isinstance(self.can_review_questions, bool):
-            raise utils.ValidationError(
-                'Expected can_review_questions to be a boolean value, '
-                'found: %s' % type(self.can_review_questions))
 
-        if not isinstance(self.can_submit_questions, bool):
-            raise utils.ValidationError(
-                'Expected can_submit_questions to be a boolean value, '
-                'found: %s' % type(self.can_submit_questions))
+class ModifiableUserDataDict(TypedDict):
+    """Dictionary representing the ModifiableUserData object."""
+
+    display_alias: str
+    pin: Optional[str]
+    preferred_language_codes: Optional[List[str]]
+    preferred_site_language_code: Optional[str]
+    preferred_audio_language_code: Optional[str]
+    user_id: Optional[str]
```

Comment:
Why are so many of these optional?

Reply:
When we are creating object from `from_dict()` method of class `ModifiableUserData`. Dict should follow the types of constructor function. that's why i added these types (optionals) according to doc string of `__init__`

Reply:
I know why `Optional` is used and what it can do. I think we should verify if all of these really need to be optional, can you please do that?

Reply:
Okay, thanks. Again please add a TODO.

Reply:
TODO is added !